v1/plugins/logs/plugin.go

```diff
@@ -372,7 +374,20 @@ func (c *Config) validateAndInjectDefaults(services []string, pluginsList []stri
 		uploadLimit = *c.Reporting.UploadSizeLimitBytes
 	}
 
-	c.Reporting.UploadSizeLimitBytes = &uploadLimit
+	switch {
+	case uploadLimit > maxUploadSizeLimitBytes:
+		maxUploadLimit := maxUploadSizeLimitBytes
+		c.Reporting.UploadSizeLimitBytes = &maxUploadLimit
+		if l != nil {
+			l.Warn("the configured `upload_size_limit_bytes` (%d) has been set to the maximum limit (%d)", uploadLimit, maxUploadLimit)
+		}
+	case uploadLimit <= minUploadSizeLimitBytes:
```

Comment:
if `uploadLimit == minUploadSizeLimitBytes`, won't that give us a pretty peculiar warning message?

```
the configured `upload_size_limit_bytes` (90) has been set to the minimum limit (90)
```

Reply:
fixed and test case added